File: C/exercise/findPoisonedDuration.c

#include "findPoisonedDuration.h"

int findPoisonedDuration(int* timeSeries, int timeSeriesSize, int duration) {
	int total = duration;
	for (int i = 1; i < timeSeriesSize; i++) {
		int diff = timeSeries[i] - timeSeries[i - 1];
		total += (diff < duration) ? diff : duration;
	}
	return total;
}

File: Ruby/solution/find_poisoned_duration.rb
# @param {Integer[]} time_series
# @param {Integer} duration
# @return {Integer}
def find_poisoned_duration(time_series, duration)
  time_series.each_cons(2).reduce(duration) do |total, (prev, curr)|
    total + [duration, curr - prev].min
  end
end
